sagemaker-python-sdk/tensorflow_script_mode_using_shell_commands/tensorflow_script_mode_using_shell_commands.ipynb

```diff
@@ -0,0 +1,303 @@
+    "The script **deep_dream.py** takes two positional arguments:\n",
+    "- `base_image_path`: Path to the image to transform.\n",
+    "- `result_prefix`: Prefix of all generated images.\n",
+    "\n",
+    "### Creating the launcher script\n",
+    "\n",
+    "We need to create a launcher script that install Keras, sets `base_image_path` \n",
+    "and `result_prefix`, and invokes **deep_dream.py**:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile launcher.sh \n",
+    "\n",
+    "pip install keras==2.2.4\n",
+    "\n",
+    "BASE_IMAGE_PATH=\"${SM_CHANNEL_TRAINING}/resting-dragon.jpg\"\n",
+    "RESULT_PREFIX=\"${SM_MODEL_DIR}/dream\"\n",
+    "\n",
+    "python deep_dream.py ${BASE_IMAGE_PATH} ${RESULT_PREFIX}\n",
+    "\n",
+    "echo \"Generated image $(ls ${SM_MODEL_DIR})\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**SM_CHANNEL_TRAINING** and **SM_MODEL** are environment variables created by the SageMaker TensorFlow\n",
+    "Container in the beggining of training. Let's take a more detailed look at then: \n",
+    "\n",
+    "- **SM_MODEL_DIR**: the directory inside the container where the training model data must be saved inside the container, i.e. /opt/ml/model.\n",
+    "- **SM_TRAINING_CHANNEL**: the directory containing data in the 'training' channel. For more information about training environment variables, please visit [SageMaker Containers](https://github.com/aws/sagemaker-containers#list-of-provided-environment-variables-by-sagemaker-containers)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Test locally using SageMaker Python SDK TensorFlow Estimator"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can use the SageMaker Python SDK [TensorFlow](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/tensorflow/README.rst#training-with-tensorflow) estimator to easily train locally and in SageMaker. To train locally, you can set the instance type to [local](https://github.com/aws/sagemaker-python-sdk#local-mode) as follow:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sagemaker\n",
+    "from sagemaker.tensorflow import TensorFlow\n",
+    "\n",
+    "estimator = TensorFlow(entry_point='launcher.sh',\n",
+    "                       dependencies=['deep_dream.py'],      # Including deep_dream.py as a dependency\n",
+    "                       train_instance_type='local',         # Run in local mode\n",
+    "                       train_instance_count=1,\n",
+    "                       role=sagemaker.get_execution_role(), # Passes to the container the AWS role that you are using on this notebook\n",
+    "                       framework_version='1.11.0',          # Uses TensorFlow 1.11\n",
```

Comment:
maybe this is because of the fix-width rendering from GH, but the inline comments take away from the readability for me. Instead, I would suggest setting these as variables where you can explain each arg - could be in the same cell or in separate ones and the explanation could be comments or markdown.